desktop/apps/auction2/components/DOM.js

```diff
@@ -0,0 +1,116 @@
+import AddToCalendar from 'desktop/components/add_to_calendar/index.coffee'
+import ClockView from 'desktop/components/clock/view.coffee'
+import ConfirmRegistrationModal from 'desktop/components/credit_card/client/confirm_registration.coffee'
+import MyActiveBids from 'desktop/components/my_active_bids/view.coffee'
+import PropTypes from 'prop-types'
+import { Component } from 'react'
+import infiniteScroll from 'desktop/apps/auction2/utils/infiniteScroll'
+import mediator from 'desktop/lib/mediator.coffee'
+import scrollToTop from 'desktop/apps/auction2/utils/scrollToTop'
+
+export default class DOM extends Component {
+  static propTypes = {
+    auction: PropTypes.object.isRequired,
+    children: PropTypes.node.isRequired,
+    me: PropTypes.object,
+    store: PropTypes.object.isRequired,
+    user: PropTypes.object
+  }
+
+  // Backbone views
+  clockView = null
+  addToCalendarView = null
+  myActiveBidsView = null
+
+  // Selectors
+  $ = null
+  $body = null
+  $registerBtn = null
+
+  componentDidMount () {
+    this.$ = require('jquery')
+    this.setupBackboneViews()
+    this.addEventListeners()
+    this.maybeShowRegistration()
+  }
+
+  componentWillUnmount () {
+    this.destroyBackboneViews()
+    this.removeEventListeners()
+  }
+
+  /**
+   * Setup Clock, Calendar and ActiveBids Backbone views.
+   * TODO: Refactor these into React components
+   */
+  setupBackboneViews () {
+    const { auction, me, user } = this.props
+    const { id, is_open, is_live_open } = auction.toJSON()
+    const showMyActiveBids = me && me.bidders.length && is_open && !is_live_open
+
+    this.clockView = new ClockView({
```

Comment:
Yup, in the component lifecycle methods, though @craigspaeth came up with a nice middleground [here](https://github.com/artsy/force/pull/1571#discussion_r123800932)